src/mkdocstrings_handlers/python/rendering.py

```diff
@@ -434,12 +436,53 @@ def do_filter_objects(
 
 
 @lru_cache(maxsize=1)
-def _get_black_formatter() -> Callable[[str, int], str]:
+def _get_formatter() -> Callable[[str, int], str]:
+    for formatter_function in [
+        _get_black_formatter,
+        _get_ruff_formatter,
+    ]:
+        if (formatter := formatter_function()) is not None:
+            return formatter
+
+    logger.info("Formatting signatures requires either Black or ruff to be installed.")
+    return lambda text, _: text
+
+
+@lru_cache(maxsize=1)
```

Comment:
If the result of `_get_formatter` is already cached, I feel like we don't have to cache both `_get_ruff_formatter` and `_get_black_formatter`. WDYT?